tools/md5sum.ml

```diff
@@ -0,0 +1,23 @@
+let get_content file =
+  let ic = open_in_bin file in
+  let buf = Buffer.create 2048 in
+  let rec fill () =
+    match input_char ic with
+    | '\r' -> fill () (* NOTE: handles the case on Windows where the
+                         git checkout has included return characters.
+                         See: https://github.com/coq/coq/pull/6305 *)
+    | c -> Buffer.add_char buf c; fill ()
+  in
+  try fill () with End_of_file -> Buffer.contents buf
```

Comment:
How about 
```ocaml
  let rec fill () =
    match input_char ic with
    | '\r' -> fill () (* NOTE: handles the case on Windows where the
                         git checkout has included return characters.
                         See: https://github.com/coq/coq/pull/6305 *)
    | c -> Buffer.add_char buf c; fill ()
    | exception End_of_file -> close_in ic; Buffer.contents buf
  in
  fill ()
```
instead?
I feel like it indicates the flow more clearly.

Reply:
It's 4.02 according to https://github.com/ocaml/ocaml/blob/150f8d2e39be9ba3dba5db1389be533387b4111f/Changes#L3499

Also, we already use that syntax in a few places and test with 4.02.3.

Reply:
Ah, right. I was looking at the wrong change comment, sorry.
Fixed.